src/backend/rest.ts

```diff
@@ -362,4 +371,42 @@ export default class RestApiBackend implements Backend {
       return new MutableDataFrame();
     }
   }
+
+  async getAutocompleteBackend(): Promise<BACKEND_TYPE> {
+    return this.api<RestApiAutocompleteResponse>({
+      url: `/objects/autocomplete/sites`,
+      method: 'POST',
+      data: { value: '', parameters: {} },
+    })
+      .then(() => {
+        return BACKEND_TYPE.REST;
+      })
+      .catch(() => {
+        return BACKEND_TYPE.WEB;
+      });
+  }
+
+  async autocompleterRequest(
+    api_url = '',
+    data: unknown
+  ): Promise<FetchResponse<WebApiResponse<ResponseDataAutocomplete>>> {
+    const { ident, params: parameters, value } = data as { ident: string; value: any; params: any };
+
+    const response = await this.api<RestApiAutocompleteResponse>({
+      url: `/objects/autocomplete/${ident}`,
+      method: 'POST',
+      data: { value, parameters },
+    });
+
+    const new_data: WebApiResponse<ResponseDataAutocomplete> = {
+      result_code: 200,
+      severity: 'success',
```

Comment:
is there a problem when the rest-api call fails? it will then result in a javascript error, because we probably can not access `response.data.choices`, right? would be okayish for me. we will revisit this with version 4 anyway?

Reply:
It should display an error on the select box who triggered the autocomplete function, but better lets catch it